client/src/nat.rs

```diff
@@ -91,9 +95,7 @@ impl<'a> NatTraverse<'a> {
         let exhausted = self.refresh_remaining()?;
 
         // Reset peer endpoints that had no viable candidates back to the server-reported one, if it exists.
-        let reset_updates = exhausted
-            .into_iter()
-            .filter_map(|peer| set_endpoint(&peer.public_key, peer.endpoint.as_ref()));
+        let reset_updates = peer_reset_updates(exhausted);
```

Comment:
Do we actually need to do this as a device update here when we're resetting them all at the end? I think it might feel cleaner if we just keep track of who needs to be reset and then do it all once the `NatTraverse` is totally done.

Reply:
Ah, do you mean we should collect them in a Vec for now and then reset them all at the end? That would work too.

I think it was originally done here because there were already other device updates happening at this point.